lib/matplotlib/axes/_axes.py

```diff
@@ -6142,6 +6147,16 @@ def _normalize_input(inp, ename='input'):
                 raise ValueError(
                     'weights should have the same shape as x')
 
+        # Combine the masks from x[i] and w[i] (if applicable) into a single
+        # mask and apply it to both.
+        if not input_empty:
+            for i in range(len(x)):
+                mask_i = x[i].mask
+                if w[i] is not None:
+                    mask_i = mask_i | w[i].mask
```

Comment:
Using histograms with weights is reasonable way to do spatial integration of images, ex `histogram(convert_pixel_locations_to_1D, bins, weights=image.ravel())`.  If you are looping over a bunch of images, some of which have artifacts that you need to mask out, being able to do this loop with out having to re-mask the pixel position values would be nice.

That said, this is something that should be pushed down into numpy?